app/controllers/users_controller.rb

```diff
@@ -20,8 +21,31 @@ def show
     @user = User.find(params[:id])
   end
 
+  def new_with_invitation_token
+    invitation = Invitation.find_by(token: params[:token])
+
+    if invitation
+      @user = User.new(email: invitation.recipient_email)
+      @invitation_token = invitation.token
+      render :new
+    else
+      redirect_to expired_token_path
+    end
+  end
+
+  private
+
   def user_params
     params.require(:user).permit(:email, :password, :full_name)
   end
 
+  def handle_invitation
+    if params[:invitation_token].present?
+      invitation = Invitation.find_by(token: params[:invitation_token])
+      @user.follow(invitation.inviter)
+      invitation.inviter.follow(@user)
+      invitation.update_column(:token, nil)
```

Comment:
This could be extracted to the instance method :) something like `#remove_token`